packages/components/src/checkbox-control/index.js

```diff
@@ -31,10 +33,15 @@ export default function CheckboxControl( {
 		} );
 	}
 
+	const ref = useRef();
 	const instanceId = useInstanceId( CheckboxControl );
 	const id = `inspector-checkbox-control-${ instanceId }`;
 	const onChangeValue = ( event ) => onChange( event.target.checked );
 
+	useEffect( () => {
+		ref.current.indeterminate = !! indeterminate;
```

Comment:
In case the `ref.current` is `null`

```suggestion
		if (ref.current) {
			ref.current.indeterminate = !! indeterminate;
		}
```